src/recommendpolicy/recommendPolicy.go

```diff
@@ -143,22 +160,46 @@ func RecommendPolicyMain() {
 			DeployNsName = append(DeployNsName, *deploy)
 		}
 
-		for _, ns := range nsNotFilter {
+		for _, ns := range NsNotFilter {
 			if d.Namespace != ns {
-				generateHardenPolicy(d.Name, d.Namespace, d.Spec.Template.Labels)
+				policy = generateHardenPolicy(d.Name, d.Namespace, d.Spec.Template.Labels)
 			}
 		}
 	}
+	for _, pol := range policy {
+		policies = append(policies, pol)
+	}
+	for _, r := range replicaSets.Items {
+		for _, ns := range NsNotFilter {
+			if r.Namespace != ns {
+				policy = generateHardenPolicy(r.Name, r.Namespace, r.Spec.Template.Labels)
+			}
+		}
+	}
+	for _, pol := range policy {
+		policies = append(policies, pol)
+	}
+	for _, r := range statefulSets.Items {
+		for _, ns := range NsNotFilter {
+			if r.Namespace != ns {
+				policy = generateHardenPolicy(r.Name, r.Namespace, r.Spec.Template.Labels)
+			}
+		}
+	}
+	for _, pol := range policy {
+		policies = append(policies, pol)
+	}
 }
 
-func generateHardenPolicy(name, namespace string, labels LabelMap) {
-	log.Info().Msgf("Generating hardening policy for deployment: %v in namespace: %v", name, namespace)
+func generateHardenPolicy(name, namespace string, labels LabelMap) []types.KnoxSystemPolicy {
+	log.Info().Msgf("Generating hardening policy for: %v in namespace: %v", name, namespace)
 	policies, err := generatePolicy(name, namespace, labels)
 	if err != nil {
 		log.Error().Msg(err.Error())
-		return
+		return nil
 	}
 	systempolicy.UpdateSysPolicies(policies)
+	return policies
```

Comment:
If we are returning the policies array from this function we can remove the `systempolicy.UpdateSysPolicies(policies)` from this function and use it in `RecommendPolicyMain()` so that it'll only get called once after we have all the policies for deployment, statefulset and replicasets.

Reply:
updated please check